src/message-queues.ts

```diff
@@ -189,6 +330,124 @@ export abstract class MessageQueue {
 
     this._options = Object.assign(defaults, options);
   }
+
+  /**
+   * Succeed a whole batch of Acks/Modacks for an OK RPC response.
+   *
+   * @private
+   */
+  handleAckSuccesses(batch: QueuedMessages) {
+    // Everyone gets a resolve!
+    batch.forEach(({responsePromise}) => {
+      responsePromise?.resolve();
+    });
+  }
+
+  /**
+   * If we get an RPC failure of any kind, this will take care of deciding
+   * what to do for each related ack/modAck. Successful ones will have their
+   * Promises resolved, permanent errors will have their Promises rejected,
+   * and transients will be returned for retry.
+   *
+   * Note that this is only used for subscriptions with exactly-once
+   * delivery enabled, so short-circuit to a success resolve on errors
+   * isn't handled here.
```

Comment:
I can't tell where this short-circuit is happening? I assume it's deeper in the stack. Should it be up top?